src/app/pages/wallets/my-wallets/my-wallet/pages-wallets-my-wallets-my-wallet.component.ts

```diff
@@ -8,4 +10,24 @@ import { MyWallet } from '../pages-wallets-my-wallet.model';
 })
 export class PagesWalletsMyWalletsMyWalletComponent {
   @Input() public wallet!: MyWallet;
-}
+
+  constructor(
+    public dialog: MatDialog,
+  ) {}
+
+  openEditWalletDialog() {
+    const dialogRef = this.dialog.open(DialogCreateOrUpdateWalletComponent, {
+        width: '350px',
+        height: '300px',
+        data: { title: 'Update Wallet', btn: "Update", inputData: this.wallet.name }
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
+        if(result) {
+            // request?
```

Comment:
After moving this to parent component you should use here `DomainsWalletsGateway`. For now we are adding in it methods doing fake request -> check for example `DomainsWalletsGateway.getWallets` how it is done for getting wallets.
You can check how payload and response for creating and updating wallets looks like here: https://bycza-zagroda.github.io/track-expenses-app-documentation/#/wallet-controller/createWallet